src/test/java/uk/gov/pay/webhooks/webhook/WebhookServiceTest.java

```diff
@@ -43,7 +51,49 @@ class WebhookServiceTest {
     
     @BeforeEach
     public void setUp() {
-        webhookService = new WebhookService(webhookDao, eventTypeDao, instantSource, idGenerator, webhookMessageDao, webhookDeliveryQueueDao);
+        webhookService = new WebhookService(webhookDao, eventTypeDao, instantSource, idGenerator, webhookMessageDao, 
+                webhookDeliveryQueueDao, webhookMessageDeletionConfig);
+    }
+    
+    @Test
+    public void shouldDeleteWebhookMessages() {
+        when(webhookMessageDeletionConfig.getMaxAgeOfMessages()).thenReturn(7);
+        when(webhookMessageDeletionConfig.getMaxNumOfMessagesToExpire()).thenReturn(2);
+        
+        var webhook = new WebhookEntity();
+        var webhookMessage1 = createWebhookMessageEntity(webhook);
+        var webhookMessage2 = createWebhookMessageEntity(webhook);
+        var webhookMessage3 = createWebhookMessageEntity(webhook);
+        var webhookMessage4 = createWebhookMessageEntity(webhook);
+        when(webhookMessageDao.getWebhookMessagesOlderThan(7)).thenReturn(
+                List.of(webhookMessage1, webhookMessage2, webhookMessage3, webhookMessage4));
```

Comment:
```suggestion
        when(webhookMessageDao.getWebhookMessagesOlderThan(7)).thenReturn(
                List<WebhookMessageEntity> webhookMessages = Collections.nCopies(4, createWebhookMessageEntity(webhook)));
```
could DRY this test up a bit?